pkg/virt-controller/watch/application.go

```diff
@@ -89,14 +104,58 @@ func (vca *VirtControllerApp) Run() {
 	stop := make(chan struct{})
 	defer close(stop)
 	vca.informerFactory.Start(stop)
-	go vca.vmController.Run(3, stop)
-	//FIXME when we have more than one worker, we need a lock on the VM
-	go vca.migrationController.Run(3, stop)
-	httpLogger := logger.With("service", "http")
-	httpLogger.Info().Log("action", "listening", "interface", vca.host, "port", vca.port)
-	if err := http.ListenAndServe(vca.host+":"+strconv.Itoa(vca.port), nil); err != nil {
+
+	run := func(stopCh <-chan struct{}) {
+		go vca.vmController.Run(3, stop)
+		//FIXME when we have more than one worker, we need a lock on the VM
+		go vca.migrationController.Run(3, stop)
+		httpLogger := logger.With("service", "http")
+		httpLogger.Info().Log("action", "listening", "interface", vca.host, "port", vca.port)
+		if err := http.ListenAndServe(vca.host+":"+strconv.Itoa(vca.port), nil); err != nil {
+			golog.Fatal(err)
+		}
+	}
+
+	recorder := createRecorder(vca.clientSet)
+
+	rl, err := resourcelock.New(resourcelock.EndpointsResourceLock,
+		"default",
+		"virt-controller",
+		vca.clientSet.CoreV1(),
+		resourcelock.ResourceLockConfig{
+			Identity:      vca.host,
```

Comment:
Host is related to the interface virt controller should bind to. It's usually going to be 0.0.0.0.

vca.host is a bad name for what it actually represents. We should change it to listening_addr or something like that.

For the Identify field here, we os.Hostname() would be a better choice.

Reply:
+1 is the same what k8s does.